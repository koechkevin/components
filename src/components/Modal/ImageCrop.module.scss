@import '../../styles/global';

.wrapper {
  height: 285px;
  position: relative;
  background-color: rgba($color: $color-black, $alpha: 0.5);

  .uploadWrap {
    flex: 1 auto;
    height: 100%;
    padding: 0;
    border-radius: 0;

    .content {
      height: 100%;
    }
  }

  .button {
    position: absolute;
    top: 70%;
    left: 5%;

    .iconBtn {
      cursor: pointer;
      color: $color-white;
    }
  }
}

:global(.ReactCrop) {
  min-width: 100%;
}

:global(.ReactCrop__crop-selection) {
  outline: none;
  pointer-events: none;
}

:global(.ReactCrop__image) {
  cursor: move;
  height: 285px;
  min-width: 100%;
  object-fit: cover;
  position: relative;
}
