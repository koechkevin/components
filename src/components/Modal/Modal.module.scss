@import '../../styles/global';

.modal {
  @include respond-to('xs-l') {
    position: fixed;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
  }

  :global {
    .ant-modal-content {
      border-radius: 8px;
      box-shadow: 0 4px 12px 0 rgba($color: $color-black, $alpha: 0.08), 0 0 0 1px $color-lighter-dark;

      @include respond-to('xs-l') {
        @include display-flex;
        flex-direction: column;
        position: fixed;
        top: 0;
        bottom: 0;
        left: 0;
        right: 0;
      }
    }

    .ant-modal-header {
      color: $color-base-dark;
      border-color: $color-black-lighter;
      border-radius: 8px 8px 0 0;
    }

    .ant-modal-close {
      width: 32px;
      height: 32px;
      top: 12px;
      right: 16px;
    }

    .ant-modal-close-x {
      width: 32px;
      height: 32px;
      line-height: 32px;
    }

    .ant-modal-title {
      font-size: 16px;
      text-align: left;
      font-weight: bold;
      line-height: 21px;
      color: $color-base-dark;
    }

    .ant-modal-body {
      padding: 32px;
      overflow: auto;
      font-size: 16px;

      @include respond-to('xs-l') {
        flex: 1;
        @include display-flex;
        @include justify-content(center);
        flex-direction: column;
      }
    }

    .ant-modal-footer {
      border: none;
      padding: 16px;
      border-radius: 0 0 8px 8px;
      background-color: $color-gray-light;

      @include respond-to('xs-l') {
        background-color: $color-white;

        .ant-row-flex {
          flex-direction: column-reverse;

          button:last-child {
            margin-left: 0;
            margin-bottom: 16px;
          }
        }
      }
    }
  }
}
