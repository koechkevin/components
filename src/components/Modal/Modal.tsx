import { faTimes } from '@fortawesome/pro-regular-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { Modal as AntModal, Row } from 'antd';
import { ButtonProps } from 'antd/lib/button';
import { ModalProps } from 'antd/lib/modal';
import React, { FC, MouseEvent, ReactNode } from 'react';
import { useMedia } from 'react-use';

import Button from '../Button/Button';
import styles from './Modal.module.scss';

interface FooterProps {
  okText?: string | ReactNode;
  okButtonProps?: ButtonProps;
  cancelText?: string | ReactNode;
  cancelButtonProps?: ButtonProps;
  onOk?: (e: MouseEvent<any>) => void;
  onCancel?: (e: MouseEvent<any>) => void;
}

export const Footer: FC<FooterProps> = (props) => {
  const { onOk, okText, cancelText, onCancel, okButtonProps, cancelButtonProps } = props;
  const isMobile = useMedia('(max-width: 575px)');

  return (
    <Row type="flex" align="middle" justify="end">
      <Button onClick={onCancel} block={isMobile} type="link" {...cancelButtonProps}>
        {cancelText || 'Cancel'}
      </Button>
      <Button type="primary" block={isMobile} size="large" onClick={onOk} {...okButtonProps}>
        {okText || 'Ok'}
      </Button>
    </Row>
  );
};

export const Modal: React.FC<ModalProps> = (props) => {
  const {
    onOk,
    okText,
    onCancel,
    className,
    children,
    cancelText,
    okButtonProps,
    confirmLoading,
    cancelButtonProps,
    ...restProps
  } = props;
  const customProps = { ...okButtonProps, loading: confirmLoading };
  const classNames = [styles.modal, className ? className : ''].join(' ');

  return (
    <AntModal
      centered
      onCancel={onCancel}
      className={classNames}
      mask={window.innerWidth > 768}
      closeIcon={<FontAwesomeIcon icon={faTimes} />}
      footer={
        <Footer
          onOk={onOk}
          okText={okText}
          onCancel={onCancel}
          cancelText={cancelText}
          okButtonProps={customProps}
          cancelButtonProps={cancelButtonProps}
        />
      }
      {...restProps}
    >
      {children}
    </AntModal>
  );
};

export default Modal;
