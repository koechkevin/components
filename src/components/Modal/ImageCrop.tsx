import { faPlus, faTrashAlt } from '@fortawesome/pro-solid-svg-icons';
import { Row } from 'antd';
import { ModalProps } from 'antd/lib/modal';
import { RcFile } from 'antd/lib/upload';
import React, { FC, useRef, useState } from 'react';
import ReactCrop from 'react-image-crop';
import { useMedia } from 'react-use';

import { getCroppedImg } from '../../utils';
import { FileDropZone } from '../FileUpload';
import { Icon } from '../Icon';
import { Tooltip } from '../Tooltip';
import Modal from './Modal';

import 'react-image-crop/dist/ReactCrop.css';
import styles from './ImageCrop.module.scss';

const accept = '.jpg, .jpeg, .png';

interface Props extends ModalProps {
  onImageUrl?: (url: any) => void;
}

export const ImageCrop: FC<Props> = (props) => {
  const { title, cancelText, width, onImageUrl, ...restProps } = props;

  const isMobile = useMedia('(max-width: 575px)');
  // const dragRef = useRef<boolean>(false);
  const imageRef = useRef<HTMLImageElement | null>(null);
  const [path, setPath] = useState<string | ArrayBuffer | null>(null);
  const [crop, setCrop] = useState<ReactCrop.Crop>({
    unit: '%',
    width: 90,
    height: 38,
    x: 5,
    y: 31,
    aspect: 536 / 108,
  });

  const beforeUpload = (file: RcFile, fileList: RcFile[]): boolean => {
    const reader = new FileReader();

    reader.onload = () => {
      setPath(reader.result);
    };
    reader.readAsDataURL(file);

    return false;
  };

  const onImageLoaded = (image: HTMLImageElement) => {
    imageRef.current = image;
  };

  const onCropChange = (crop: ReactCrop.Crop, percentCrop: ReactCrop.PercentCrop) => {
    setCrop(percentCrop);
  };

  const makeClientCrop = async () => {
    if (imageRef && imageRef.current && crop.width && crop.height) {
      const croppedImageUrl = await getCroppedImg(imageRef.current, crop);
      onImageUrl && onImageUrl(croppedImageUrl);
    }
  };

  const onRemove = () => setPath(null);

  return (
    <Modal width={width} title="Edit Image" okText="Save Changes" bodyStyle={{ padding: 0 }} {...restProps}>
      <Row type="flex" justify="center" align="middle" className={styles.wrapper}>
        {!path ? (
          <FileDropZone
            accept={accept}
            beforeUpload={beforeUpload}
            wrapClassName={styles.uploadWrap}
            contentClassName={styles.content}
            buttonText={isMobile ? 'Upload image' : 'Drag and drop or upload'}
          />
        ) : (
          <>
            <ReactCrop
              locked
              disabled
              crop={crop}
              ruleOfThirds
              onChange={onCropChange}
              onImageLoaded={onImageLoaded}
              src={typeof path === 'string' ? path : ''}
            />
            <Row className={styles.button}>
              <Tooltip title="Remove" light>
                <Icon icon={faTrashAlt} onClick={onRemove} className={styles.iconBtn} />
              </Tooltip>
              <Tooltip title="Crop" light>
                <Icon icon={faPlus} onClick={makeClientCrop} className={styles.iconBtn} />
              </Tooltip>
            </Row>
          </>
        )}
      </Row>
    </Modal>
  );
};

ImageCrop.defaultProps = {
  width: 600,
};

export default ImageCrop;
