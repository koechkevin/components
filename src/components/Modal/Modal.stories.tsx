import { boolean } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import React from 'react';

import { Center } from '../Center';
import ImageCrop from './ImageCrop';
import Modal from './Modal';

const visible = boolean('visible', true);
const stories = storiesOf('Modal', module);

stories
  .addDecorator((story) => <Center>{story()}</Center>)
  .add(
    'Default modal',
    () => (
      <Modal visible={visible} title="Modal Example">
        Modal Content
      </Modal>
    ),
    {
      notes: 'A modal component customized with Antd Modal, the component props is the same with Ant ModalProps',
    },
  )
  .add('Modal without title and footer', () => (
    <Modal visible={visible} title={null} footer={null}>
      Modal Content
    </Modal>
  ))
  .add('Modal without footer', () => (
    <Modal visible={visible} title="Modal Example" footer={null}>
      Modal Content
    </Modal>
  ))
  .add('Image crop modal', () => <ImageCrop visible={visible} />);
