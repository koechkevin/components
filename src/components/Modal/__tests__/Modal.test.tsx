import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';

import Modal, { Footer } from '../Modal';

describe('Modal Test', () => {
  it('should not display the modal', () => {
    const tree = shallow(<Modal visible={false}>Modal Test</Modal>);
    const props = tree.props();

    expect(props.visible).toEqual(false);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should display the modal', () => {
    const tree = shallow(<Modal visible>Modal Test</Modal>);
    const props = tree.props();

    expect(props.visible).toEqual(true);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should display the modal without title', () => {
    const tree = shallow(
      <Modal visible title={null}>
        Modal Test
      </Modal>,
    );
    const props = tree.props();

    expect(props.title).toEqual(null);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should display the modal with title "Test Modal Title"', () => {
    const tree = shallow(
      <Modal visible title="Test Modal Title">
        Modal Test
      </Modal>,
    );
    const props = tree.props();

    expect(props.title).toEqual('Test Modal Title');
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should display the modal without footer', () => {
    const tree = shallow(
      <Modal visible footer={null}>
        Modal Test
      </Modal>,
    );
    const props = tree.props();

    expect(props.footer).toEqual(null);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should display the modal with footer "Test Modal Footer"', () => {
    const tree = shallow(
      <Modal visible footer="Test Modal Footer">
        Modal Test
      </Modal>,
    );
    const props = tree.props();

    expect(props.footer).toEqual('Test Modal Footer');
    expect(toJson(tree)).toMatchSnapshot();
  });
});

describe('Modal Footer Test', () => {
  it('should display the footer with a title "Ok Button"', () => {
    const tree = mount(<Footer okText="Ok Button">Modal Test</Footer>);
    const props = tree.props();

    expect(props.okText).toEqual('Ok Button');
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should display the footer with cancel button text "Cancel Button"', () => {
    const tree = mount(<Footer cancelText="Cancel Button">Modal Test</Footer>);
    const props = tree.props();

    expect(props.cancelText).toEqual('Cancel Button');
    expect(toJson(tree)).toMatchSnapshot();
  });
});
