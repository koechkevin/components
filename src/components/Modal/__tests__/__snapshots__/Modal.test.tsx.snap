// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Footer Test should display the footer with a title "Ok Button" 1`] = `
<Footer
  okText="Ok Button"
>
  <Row
    align="middle"
    gutter={0}
    justify="end"
    type="flex"
  >
    <div
      className="ant-row-flex ant-row-flex-end ant-row-flex-middle"
      style={Object {}}
    >
      <Button
        block={false}
        type="link"
      >
        <Button
          block={false}
          className="button  "
          ghost={false}
          htmlType="button"
          loading={false}
          type="link"
        >
          <button
            className="ant-btn button   ant-btn-link"
            onClick={[Function]}
            type="button"
          >
            <span
              key=".0"
            >
              Cancel
            </span>
          </button>
        </Button>
      </Button>
      <Button
        block={false}
        size="large"
        type="primary"
      >
        <Button
          block={false}
          className="button  "
          ghost={false}
          htmlType="button"
          loading={false}
          size="large"
          type="primary"
        >
          <Wave>
            <button
              className="ant-btn button   ant-btn-primary ant-btn-lg"
              onClick={[Function]}
              type="button"
            >
              <span
                key=".0"
              >
                Ok Button
              </span>
            </button>
          </Wave>
        </Button>
      </Button>
    </div>
  </Row>
</Footer>
`;

exports[`Modal Footer Test should display the footer with cancel button text "Cancel Button" 1`] = `
<Footer
  cancelText="Cancel Button"
>
  <Row
    align="middle"
    gutter={0}
    justify="end"
    type="flex"
  >
    <div
      className="ant-row-flex ant-row-flex-end ant-row-flex-middle"
      style={Object {}}
    >
      <Button
        block={false}
        type="link"
      >
        <Button
          block={false}
          className="button  "
          ghost={false}
          htmlType="button"
          loading={false}
          type="link"
        >
          <button
            className="ant-btn button   ant-btn-link"
            onClick={[Function]}
            type="button"
          >
            <span
              key=".0"
            >
              Cancel Button
            </span>
          </button>
        </Button>
      </Button>
      <Button
        block={false}
        size="large"
        type="primary"
      >
        <Button
          block={false}
          className="button  "
          ghost={false}
          htmlType="button"
          loading={false}
          size="large"
          type="primary"
        >
          <Wave>
            <button
              className="ant-btn button   ant-btn-primary ant-btn-lg"
              onClick={[Function]}
              type="button"
            >
              <span
                key=".0"
              >
                Ok
              </span>
            </button>
          </Wave>
        </Button>
      </Button>
    </div>
  </Row>
</Footer>
`;

exports[`Modal Test should display the modal 1`] = `
<Modal
  centered={true}
  className="modal "
  closeIcon={
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            320,
            512,
            Array [],
            "f00d",
            "M207.6 256l107.72-107.72c6.23-6.23 6.23-16.34 0-22.58l-25.03-25.03c-6.23-6.23-16.34-6.23-22.58 0L160 208.4 52.28 100.68c-6.23-6.23-16.34-6.23-22.58 0L4.68 125.7c-6.23 6.23-6.23 16.34 0 22.58L112.4 256 4.68 363.72c-6.23 6.23-6.23 16.34 0 22.58l25.03 25.03c6.23 6.23 16.34 6.23 22.58 0L160 303.6l107.72 107.72c6.23 6.23 16.34 6.23 22.58 0l25.03-25.03c6.23-6.23 6.23-16.34 0-22.58L207.6 256z",
          ],
          "iconName": "times",
          "prefix": "far",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  }
  confirmLoading={false}
  footer={
    <Footer
      okButtonProps={
        Object {
          "loading": undefined,
        }
      }
    />
  }
  mask={true}
  maskTransitionName="fade"
  okType="primary"
  transitionName="zoom"
  visible={true}
  width={520}
>
  Modal Test
</Modal>
`;

exports[`Modal Test should display the modal with footer "Test Modal Footer" 1`] = `
<Modal
  centered={true}
  className="modal "
  closeIcon={
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            320,
            512,
            Array [],
            "f00d",
            "M207.6 256l107.72-107.72c6.23-6.23 6.23-16.34 0-22.58l-25.03-25.03c-6.23-6.23-16.34-6.23-22.58 0L160 208.4 52.28 100.68c-6.23-6.23-16.34-6.23-22.58 0L4.68 125.7c-6.23 6.23-6.23 16.34 0 22.58L112.4 256 4.68 363.72c-6.23 6.23-6.23 16.34 0 22.58l25.03 25.03c6.23 6.23 16.34 6.23 22.58 0L160 303.6l107.72 107.72c6.23 6.23 16.34 6.23 22.58 0l25.03-25.03c6.23-6.23 6.23-16.34 0-22.58L207.6 256z",
          ],
          "iconName": "times",
          "prefix": "far",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  }
  confirmLoading={false}
  footer="Test Modal Footer"
  mask={true}
  maskTransitionName="fade"
  okType="primary"
  transitionName="zoom"
  visible={true}
  width={520}
>
  Modal Test
</Modal>
`;

exports[`Modal Test should display the modal with title "Test Modal Title" 1`] = `
<Modal
  centered={true}
  className="modal "
  closeIcon={
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            320,
            512,
            Array [],
            "f00d",
            "M207.6 256l107.72-107.72c6.23-6.23 6.23-16.34 0-22.58l-25.03-25.03c-6.23-6.23-16.34-6.23-22.58 0L160 208.4 52.28 100.68c-6.23-6.23-16.34-6.23-22.58 0L4.68 125.7c-6.23 6.23-6.23 16.34 0 22.58L112.4 256 4.68 363.72c-6.23 6.23-6.23 16.34 0 22.58l25.03 25.03c6.23 6.23 16.34 6.23 22.58 0L160 303.6l107.72 107.72c6.23 6.23 16.34 6.23 22.58 0l25.03-25.03c6.23-6.23 6.23-16.34 0-22.58L207.6 256z",
          ],
          "iconName": "times",
          "prefix": "far",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  }
  confirmLoading={false}
  footer={
    <Footer
      okButtonProps={
        Object {
          "loading": undefined,
        }
      }
    />
  }
  mask={true}
  maskTransitionName="fade"
  okType="primary"
  title="Test Modal Title"
  transitionName="zoom"
  visible={true}
  width={520}
>
  Modal Test
</Modal>
`;

exports[`Modal Test should display the modal without footer 1`] = `
<Modal
  centered={true}
  className="modal "
  closeIcon={
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            320,
            512,
            Array [],
            "f00d",
            "M207.6 256l107.72-107.72c6.23-6.23 6.23-16.34 0-22.58l-25.03-25.03c-6.23-6.23-16.34-6.23-22.58 0L160 208.4 52.28 100.68c-6.23-6.23-16.34-6.23-22.58 0L4.68 125.7c-6.23 6.23-6.23 16.34 0 22.58L112.4 256 4.68 363.72c-6.23 6.23-6.23 16.34 0 22.58l25.03 25.03c6.23 6.23 16.34 6.23 22.58 0L160 303.6l107.72 107.72c6.23 6.23 16.34 6.23 22.58 0l25.03-25.03c6.23-6.23 6.23-16.34 0-22.58L207.6 256z",
          ],
          "iconName": "times",
          "prefix": "far",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  }
  confirmLoading={false}
  footer={null}
  mask={true}
  maskTransitionName="fade"
  okType="primary"
  transitionName="zoom"
  visible={true}
  width={520}
>
  Modal Test
</Modal>
`;

exports[`Modal Test should display the modal without title 1`] = `
<Modal
  centered={true}
  className="modal "
  closeIcon={
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            320,
            512,
            Array [],
            "f00d",
            "M207.6 256l107.72-107.72c6.23-6.23 6.23-16.34 0-22.58l-25.03-25.03c-6.23-6.23-16.34-6.23-22.58 0L160 208.4 52.28 100.68c-6.23-6.23-16.34-6.23-22.58 0L4.68 125.7c-6.23 6.23-6.23 16.34 0 22.58L112.4 256 4.68 363.72c-6.23 6.23-6.23 16.34 0 22.58l25.03 25.03c6.23 6.23 16.34 6.23 22.58 0L160 303.6l107.72 107.72c6.23 6.23 16.34 6.23 22.58 0l25.03-25.03c6.23-6.23 6.23-16.34 0-22.58L207.6 256z",
          ],
          "iconName": "times",
          "prefix": "far",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  }
  confirmLoading={false}
  footer={
    <Footer
      okButtonProps={
        Object {
          "loading": undefined,
        }
      }
    />
  }
  mask={true}
  maskTransitionName="fade"
  okType="primary"
  title={null}
  transitionName="zoom"
  visible={true}
  width={520}
>
  Modal Test
</Modal>
`;

exports[`Modal Test should not display the modal 1`] = `
<Modal
  centered={true}
  className="modal "
  closeIcon={
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            320,
            512,
            Array [],
            "f00d",
            "M207.6 256l107.72-107.72c6.23-6.23 6.23-16.34 0-22.58l-25.03-25.03c-6.23-6.23-16.34-6.23-22.58 0L160 208.4 52.28 100.68c-6.23-6.23-16.34-6.23-22.58 0L4.68 125.7c-6.23 6.23-6.23 16.34 0 22.58L112.4 256 4.68 363.72c-6.23 6.23-6.23 16.34 0 22.58l25.03 25.03c6.23 6.23 16.34 6.23 22.58 0L160 303.6l107.72 107.72c6.23 6.23 16.34 6.23 22.58 0l25.03-25.03c6.23-6.23 6.23-16.34 0-22.58L207.6 256z",
          ],
          "iconName": "times",
          "prefix": "far",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  }
  confirmLoading={false}
  footer={
    <Footer
      okButtonProps={
        Object {
          "loading": undefined,
        }
      }
    />
  }
  mask={true}
  maskTransitionName="fade"
  okType="primary"
  transitionName="zoom"
  visible={false}
  width={520}
>
  Modal Test
</Modal>
`;
